src/Core/Form/ChoiceProvider/ProductImagesChoiceProvider.php

```diff
@@ -59,7 +74,10 @@ public function getChoices(array $options): array
         }
 
         /** @var ProductImage[] $productImages */
-        $productImages = $this->queryBus->handle(new GetProductImages((int) $options['product_id']));
+        $productImages = $this->queryBus->handle(new GetProductImages(
```

Comment:
This choice provider is actually used in the combination form, it list all the images of the product so that the combination can select the appropriate ones. It's not to be handled right now, but I think we'll have to think about how the multishop selection is handled in the combination form @MatShir

Reply:
But anyway @lartist as I commented above, for now it seems safer to always return the whole list of product images regardless of the provided shop constraint (the constraint should impact the content for cover only)